src/main/java/bookface/model/person/UniquePersonList.java

```diff
@@ -72,13 +73,16 @@ public void setPerson(Person target, Person editedPerson) {
     }
 
     /**
-     * Returns all books that are loaned by the user.
-     * The person must exist in the list.
+     * Removes the person from the list. The person must exist in the list.
+     * If the {@code Person} has any {@code Book} on loan, {@code Person} will
+     * not be deleted and an exception will be thrown
+     *
+     * @param toRemove Person to remove
      */
     public void remove(Person toRemove) {
         requireNonNull(toRemove);
-        for (Book book : toRemove.getLoanedBooksSet()) {
-            book.markBookAsReturned();
+        if (toRemove.hasBooksOnLoan()) {
```

Comment:
We can just remove this, as we already check if a person has loaned books in the Command itself - then we also don't need to refresh the UI after a delete. I'll help simplify the code for you